src/main/java/ch/wisv/areafiftylan/web/tournament/controller/TournamentController.java

```diff
@@ -42,6 +43,29 @@ public TournamentController(TournamentService tournamentService) {
         return createResponseEntity(HttpStatus.CREATED, "Tournament added.", tournament);
     }
 
+    @PostMapping("/{id}")
+    @PreAuthorize("hasRole('COMMITTEE')")
+    ResponseEntity<?> editTournament(@RequestBody Tournament tournament) {
+        Tournament dbTournament = tournamentService.getTournaments()
+                .stream()
+                .filter(tournament1 -> tournament1.getId().equals(tournament.getId()))
+                .findFirst()
+                .orElseThrow(TournamentNotFoundException::new);
+
+        // This is a bit double, but it ensures we update an existing tournament
```

Comment:
Is there a field we're not updating here? If you're literally replacing them all, you can just replace it with the new responsebody object. Also, this is typical service logic, shouldn't be in the controller

Reply:
Gonna be honest, I was pretty drunk when I wrote this :upside_down_face: 

I fixed it now ;)